### PR TITLE
Update loop docs

### LINKED ISSUE #53: loop speedups
@patricksurry your loop speedup work so far has made Tali noticeably faster.  I've added cycle counting with a few simple looping scenarios, and here are the before/after results:
```
Before merging loop speedup:
: do?word1 5 5 ?do loop ;  ok
             ' do?word1      cycle_test            CYCLES:    322 ok
: do?word2 100 0 ?do i drop loop ;  ok
             ' do?word2      cycle_test            CYCLES:  12836 ok
: doword 100 0 do loop ;  ok
             ' doword        cycle_test            CYCLES:   6700 ok
: dowordi 100 0 do i drop loop ;  ok
             ' dowordi       cycle_test            CYCLES:  12700 ok
: dodoword 100 0 do 10 0 do loop loop ;  ok
             ' dodoword      cycle_test            CYCLES:  90500 ok
: dodowordij 100 0 do 10 0 do i drop j drop loop loop ;  ok
             ' dodowordij    cycle_test            CYCLES: 210500 ok
: doword+loop 100 0 do 5 +loop ;  ok
             ' doword+loop   cycle_test            CYCLES:   2420 ok

After merging loop speedup:
: do?word1 5 5 ?do loop ;  ok
             ' do?word1      cycle_test            CYCLES:    322 ok
: do?word2 100 0 ?do i drop loop ;  ok
             ' do?word2      cycle_test            CYCLES:   8384 ok
: doword 100 0 do loop ;  ok
             ' doword        cycle_test            CYCLES:   2148 ok
: dowordi 100 0 do i drop loop ;  ok
             ' dowordi       cycle_test            CYCLES:   8248 ok
: dodoword 100 0 do 10 0 do loop loop ;  ok
             ' dodoword      cycle_test            CYCLES:  44748 ok
: dodowordij 100 0 do 10 0 do i drop j drop loop loop ;  ok
             ' dodowordij    cycle_test            CYCLES: 165748 ok
: doword+loop 100 0 do 5 +loop ;  ok
             ' doword+loop   cycle_test            CYCLES:   2418 ok
```

### PR DESCRIPTION
**Pull request by patricksurry**
Fixes #53

Updates a few comments in the code and fixes the glossary which I broke with the native_words split.

No functional change.

### CODE REVIEW

File: docs/ch_glossary.adoc [hunk 1]
```diff
@@ -555,7 +555,7 @@ https://forth-standard.org/standard/tools/DUMP
 `dup`:: _ANS core_ ( u -- u u ) "Duplicate TOS"
 https://forth-standard.org/standard/core/DUP
 
-`ed`:: _Tali Forth_ ( -- u ) "Line-based editor"
+`ed:`:: _Tali Forth_ ( -- u ) "Line-based editor"
```
Comment (SamCoVT) [inline docs/ch_glossary.adoc:558]: Oh, thanks!  I missed that one.

File: words/core.asm [hunk 2]
```diff
@@ -7302,7 +7300,7 @@ z_um_star:      rts
 
 
 
-; ## UNLOOP ( -- )(R: n1 n2 n3 ---) "Drop loop control from Return stack"
+; ## UNLOOP ( -- )(R: n1 n2 n3 ---) "Drop current loop control block"
```
Comment (SamCoVT) [inline words/core.asm]: You can remove the return stack effects on this one too, can't you?

Comment (patricksurry) [reply words/core.asm]: good point, there are a few like that I just cleaned up

**Comment by SamCoVT** [comment]
Is this one ready to merge?

**Comment by patricksurry** [comment]
yup

**Pull request merged**